### DISCUSSION TITLE
React beautiful drag and drop and useFieldArray issues

### DISCUSSION THREAD

**Discussion by wdfinch** [react-hook-form/react-hook-form#3998] [General]
Right now it is quite difficult to use react beautiful drag and drop and `useFieldArray`. It seems like the main issue is react beautiful drag and drop relies on each field item having a consistent id, however, the fields being dragged can be unmounted/moved via a portal during the drag and it seems like the id is therefore not consistent during the whole process (even when using `shouldUnregister: false`).

Given that this is such a popular library and the act of reordering a field array is common (even without this lib), I'm wondering how we can work together to make this easier or provide better docs on this subject. The closest docs are [Working with virtualized lists](https://react-hook-form.com/advanced-usage#Workingwithvirtualizedlists), but it seems like it doesn't apply to react beautiful drag and drop (especially when moving the dragged item into a portal).

Can someone provide some details on how V7 will affect this and what we can do here? I'd be happy to provide a sandbox to show some of the challenges. 

#3981

**Comment by bluebill1049** [comment]
hey, @wdfinch we should be able to ship an alpha version next week, I think it's worth testing it out.

**Reply by wdfinch** [reply]
Thanks! I've been really struggling on this one.

**Comment by RwwL** [comment]
It's almost a year later now, but I'm curious if anything ever came of this and/or if anyone has a great example of using `react-beautiful-dnd` with `useFieldArray`.

**Comment by charlestbell** [comment]
Stumbled on to this: https://codesandbox.io/s/bzjwz?file=/src/index.js

**Reply by 0xMALVEE** [reply]
thanks +rep
